src/Core/Services/TypeHelper.cs

```diff
@@ -2,120 +2,202 @@
+        [typeof(ushort?)] = DbType.UInt16,
+        [typeof(int?)] = DbType.Int32,
+        [typeof(uint?)] = DbType.UInt32,
+        [typeof(long?)] = DbType.Int64,
+        [typeof(ulong?)] = DbType.UInt64,
+        [typeof(float?)] = DbType.Single,
+        [typeof(double?)] = DbType.Double,
+        [typeof(decimal?)] = DbType.Decimal,
+        [typeof(bool?)] = DbType.Boolean,
+        [typeof(char?)] = DbType.StringFixedLength,
+        [typeof(Guid?)] = DbType.Guid,
+        [typeof(object)] = DbType.Object
+    };
+
+    /// <summary>
+    /// Maps .NET Framework type (System/CLR type) to JsonDataType.
+    /// </summary>
+    private static Dictionary<Type, JsonDataType> _systemTypeToJsonDataTypeMap = new()
+    {
+        [typeof(byte)] = JsonDataType.String,
+        [typeof(sbyte)] = JsonDataType.String,
+        [typeof(short)] = JsonDataType.Number,
+        [typeof(ushort)] = JsonDataType.Number,
+        [typeof(int)] = JsonDataType.Number,
+        [typeof(uint)] = JsonDataType.Number,
+        [typeof(long)] = JsonDataType.Number,
+        [typeof(ulong)] = JsonDataType.Number,
+        [typeof(float)] = JsonDataType.Number,
+        [typeof(double)] = JsonDataType.Number,
+        [typeof(decimal)] = JsonDataType.Number,
+        [typeof(bool)] = JsonDataType.Boolean,
+        [typeof(string)] = JsonDataType.String,
+        [typeof(char)] = JsonDataType.String,
+        [typeof(Guid)] = JsonDataType.String,
+        [typeof(byte[])] = JsonDataType.String,
+        [typeof(TimeSpan)] = JsonDataType.String,
+        [typeof(byte?)] = JsonDataType.String,
+        [typeof(sbyte?)] = JsonDataType.String,
+        [typeof(short?)] = JsonDataType.Number,
+        [typeof(ushort?)] = JsonDataType.Number,
+        [typeof(int?)] = JsonDataType.Number,
+        [typeof(uint?)] = JsonDataType.Number,
+        [typeof(long?)] = JsonDataType.Number,
+        [typeof(ulong?)] = JsonDataType.Number,
+        [typeof(float?)] = JsonDataType.Number,
+        [typeof(double?)] = JsonDataType.Number,
+        [typeof(decimal?)] = JsonDataType.Number,
+        [typeof(bool?)] = JsonDataType.Boolean,
+        [typeof(char?)] = JsonDataType.String,
+        [typeof(Guid?)] = JsonDataType.String,
+        [typeof(object)] = JsonDataType.Object,
+        [typeof(DateTime)] = JsonDataType.String,
+        [typeof(DateTimeOffset)] = JsonDataType.String
+    };
 
-        /// <summary>
-        /// Returns the DbType for given system type.
-        /// </summary>
-        /// <param name="systemType">The system type for which the DbType is to be determined.</param>
-        /// <returns>DbType for the given system type.</returns>
-        public static DbType? GetDbTypeFromSystemType(Type systemType)
+    /// <summary>
+    /// Maps SqlDbType enum to .NET Framework type (System type).
+    /// </summary>
+    private static readonly Dictionary<SqlDbType, Type> _sqlDbTypeToType = new()
+    {
+        { SqlDbType.BigInt,typeof(long) },
```

Comment:
The way this dictionary is defined is different to the ones above (they're using indexer assignment rather than providing an enumable of key/value pairs).

Should we have one pattern used consistently?

Reply:
Updated for consistency to be defined by providing an enumerable of key/value pairs.